go/gas-oracle/oracle/updater_interface_test.go

```diff
@@ -0,0 +1,130 @@
+package oracle
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/go/gas-oracle/bindings"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethdb"
+)
+
+func TestWrapGetLatestBlockNumberFn(t *testing.T) {
+	key, _ := crypto.GenerateKey()
+	sim, db := newSimulatedBackend(key)
+	chain := sim.Blockchain()
+
+	getLatest := wrapGetLatestBlockNumberFn(sim)
+
+	// Generate a valid chain of 10 blocks
+	blocks, _ := core.GenerateChain(chain.Config(), chain.CurrentBlock(), chain.Engine(), db, 10, nil)
+
+	// Check that the latest is 0 to start
+	latest, err := getLatest()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if latest != 0 {
+		t.Fatal("not zero")
+	}
+
+	// Insert the blocks one by one and assert that they are incrementing
+	for i, block := range blocks {
+		if _, err := chain.InsertChain([]*types.Block{block}); err != nil {
+			t.Fatal(err)
+		}
+		latest, err := getLatest()
+		if err != nil {
+			t.Fatal(err)
+		}
+		// Handle zero index by adding 1
+		if latest != uint64(i+1) {
+			t.Fatal("mismatch")
+		}
+	}
+}
+
+func TestWrapUpdateL2GasPriceFn(t *testing.T) {
```

Comment:
For completeness sake, you could also test that the "change is significant" logic works properly. Presumably you'd add this test with something like:
- set gas price 1
- attempt set gas price 1.1 with a 20% threshold (or something)
- check that no new transactions were sent to the simulated backend

Is it easy to check if there was a transaction sent to the simulated chain after the function call?

Reply:
I can add a test for that, I did write a unit test `TestIsDifferenceSignificant` that tests the change is significant logic itself